apps/namadillo/src/App/Staking/IncrementBonding.tsx

```diff
@@ -83,12 +84,13 @@ const IncrementBonding = (): JSX.Element => {
       "Extension is not connected or you don't have an account"
     );
     invariant(gasPrice.data, "Gas price loading is still pending");
+    invariant(gasLimits.isSuccess, "Gas limit loading is still pending");
     createBondTransaction({
       changes: parseUpdatedAmounts(),
       account,
       gasConfig: {
         gasPrice: gasPrice.data!,
-        gasLimit: GAS_LIMIT,
+        gasLimit: gasLimits.data!.Bond.native,
```

Comment:
Maybe this approach will not be sufficient for batch transactions :( What do you think?